solver/src/driver/solver_settlements.rs

```diff
@@ -79,11 +79,19 @@ pub struct RatedSettlement {
 }
 
 impl RatedSettlement {
-    pub fn objective_value(&self, gas_price: f64) -> BigRational {
-        let gas_price = BigRational::from_float(gas_price).unwrap();
+    fn compute_objective_value(
+        surplus: &BigRational,
+        gas_estimate: &BigRational,
+        gas_price_normalized: &BigRational,
+    ) -> BigRational {
+        let cost = gas_estimate * gas_price_normalized;
+        surplus - cost
+    }
+
+    // gas_price_normalized is the price of gas normalized to the found price vector.
+    pub fn objective_value(&self, gas_price_normalized: &BigRational) -> BigRational {
         let gas_estimate = self.gas_estimate.to_big_rational();
-        let cost = gas_estimate * gas_price;
-        self.surplus.clone() - cost
+        Self::compute_objective_value(&self.surplus, &gas_estimate, gas_price_normalized)
```

Comment:
> giving the function on the settlement struct (which currently computes the surplus) also access to the gas estimate

It already has access to the gas_estimate. Do you mean the gas_price_normalized?

Reply:
I think it might be nice to unit test the objective function via some thing that takes:
1. A set of computed prices and trades (e.g. a settlement)
2. The pre-auction prices (for normalization)
3. The current gas price in wei
4. The amount of gas needed for the transaction

And then do some assertions on that holistic computation (what is the objective value given certain individual utilities), at what gas price does a "bad but cheap" solution become higher ranked than another "good but expensive" solution, etc.

This way everything gets computed in one place rather than carrying around intermediate results (like surplus) on helper structs, which can easily lead to confusing the unit of account (and also using the wrong unit of account inside the unit tests).

Reply:
Refactored below to address this. Still kept the helper function to allow to unit test just the final formula.